lib/ansible/module_utils/mysql.py

```diff
@@ -75,6 +76,9 @@ def mysql_connect(module, login_user=None, login_password=None, config_file='',
 
     db_connection = mysql_driver.connect(**config)
     if cursor_class is not None:
-        return db_connection.cursor(cursorclass=mysql_driver.cursors.DictCursor)
```

Comment:
I don't really understand why `mysql_driver.cursors.DictCursor` is hardcoded here. `mysql_replication` passes `'mysql_driver.cursors.DictCursor'` as `cursor_class` (and it is the only `mysql_*` module passing it), while most `proxysql_*` modules pass `cursor_class=mysql_driver.cursors.DictCursor` (and one or two doesn't pass it at all).

Either this should be named `use_dict_cursor` (or something like that), or it should be passed to `db_connection.cursor()` directly (and the call in `mysql_replication` should be changed). That's nothing you have to do for this PR, though, it's just a general rant after trying to understand what's going on :-)

Reply:
Yeah, I had the same problems trying to understand this; it's not really making sense but I just wanted to fix the problem which I've been facing with this cuserclass madness... :)